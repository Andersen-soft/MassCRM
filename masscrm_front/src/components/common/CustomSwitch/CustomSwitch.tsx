import React, { FC } from 'react';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';
import { ICustomSwitch } from './interfaces';
import { switchStyle } from './CustomSwitch.style';

export const CustomSwitch: FC<ICustomSwitch> = ({
  label,
  value,
  onChangeHandler,
  name
}) => {
  const style = switchStyle();

  return (
    <FormControlLabel
      classes={{ root: style.switch }}
      label={label}
      control={
        <Switch onChange={onChangeHandler} checked={value} name={name} />
      }
      labelPlacement='start'
    />
  );
};
