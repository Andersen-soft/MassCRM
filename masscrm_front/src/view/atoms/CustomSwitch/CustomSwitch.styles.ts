import { createStyles, makeStyles } from '@material-ui/core';

export const useStyles = makeStyles(() =>
  createStyles({
    switch: {
      color: '#939BB2',
      width: 'auto',
      height: 'auto',
      '& .MuiSwitch-root': {
        padding: '9px'
      },
      '& .MuiTypography-body1': {
        fontSize: '0.85em'
      },
      '& .MuiSwitch-track': {
        width: '36px',
        height: '20px',
        borderRadius: '20px',
        backgroundColor: 'rgba(105, 115, 143, 0.5)'
      },
      '& .MuiSwitch-colorSecondary.Mui-checked': {
        color: '#fff',
        transform: 'translateX(15px)'
      },
      '& .MuiSwitch-colorSecondary.Mui-checked + .MuiSwitch-track': {
        backgroundColor: '#FEDA00'
      },
      '& .MuiSwitch-thumb': {
        width: '16px',
        height: '16px'
      },
      '& .MuiSwitch-switchBase': {
        margin: '9px 10px',
        padding: '2px'
      }
    }
  })
);

export const disabledStyles = makeStyles(() => ({
  disabled: {
    cursor: 'default',
    '&:hover': {
      backgroundColor: 'inherit'
    }
  }
}));
