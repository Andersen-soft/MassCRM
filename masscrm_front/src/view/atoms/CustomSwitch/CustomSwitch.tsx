import React, { FC } from 'react';
import { Switch, FormControlLabel } from '@material-ui/core';
import { useStyles, disabledStyles } from './CustomSwitch.styles';

interface IProps {
  onChangeHandler?: (event?: any) => void;
  value: boolean;
  label?: string;
  name?: string;
  disabled?: boolean;
}

export const CustomSwitch: FC<IProps> = ({
  label,
  value,
  onChangeHandler,
  name,
  disabled
}) => {
  const styles = useStyles();
  const disabledClasses = disabledStyles();

  return (
    <FormControlLabel
      classes={{ root: styles.switch }}
      label={label}
      control={
        <Switch
          classes={{ switchBase: disabled ? disabledClasses.disabled : '' }}
          onChange={onChangeHandler}
          checked={value}
          name={name}
        />
      }
      labelPlacement='start'
    />
  );
};
