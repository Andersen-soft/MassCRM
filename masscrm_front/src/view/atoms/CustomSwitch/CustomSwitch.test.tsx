import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { CustomSwitch } from './CustomSwitch';

describe('CustomSwitch', () => {
  const mockCallBack = jest.fn();

  const switchBtn = shallow(
    <CustomSwitch onChangeHandler={mockCallBack} value />
  );

  it('CustomSwitch should render correctly', () => {
    expect(shallowToJson(switchBtn)).toMatchSnapshot();
  });
});
