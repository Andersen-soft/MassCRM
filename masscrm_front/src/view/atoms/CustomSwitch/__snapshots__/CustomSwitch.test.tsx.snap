// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSwitch CustomSwitch should render correctly 1`] = `
<WithStyles(ForwardRef(FormControlLabel))
  classes={
    Object {
      "root": "makeStyles-switch-1",
    }
  }
  control={
    <ForwardRef(WithStyles)
      checked={true}
      classes={
        Object {
          "switchBase": "",
        }
      }
      onChange={[MockFunction]}
    />
  }
  labelPlacement="start"
/>
`;
